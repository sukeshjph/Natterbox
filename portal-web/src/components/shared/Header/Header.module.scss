.root {
  background-color: #5c6770;
}

.upperHeader {
  min-height: 80px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.lowerHeader {
  display: flex;
  justify-content: space-between;
}

.rectangle {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 116px;
  height: 40px;
  border-radius: 4px;
  border: solid 1px #c4d600;
  background-color: #c4d600;
  margin-left: 25px;
}

.search {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-right: 10px;
  margin-bottom: 5px;
  min-width: 250px;
}

.tabLink {
  span:first-child {
    font-size: 18px;
    font-weight: 500;
    text-transform: none;
  }
}

.tabLinkSelected {
  border-bottom: 2px solid white;
}
