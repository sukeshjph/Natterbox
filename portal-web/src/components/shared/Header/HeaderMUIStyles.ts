import { makeStyles } from "@material-ui/core/styles"

export const tabsStyles = makeStyles({
  root: {
    paddingLeft: "10px",
  },
})

export const tabStyles = makeStyles({
  root: {
    fontSize: "1.25rem",
    fontWeight: 600,
    maxWidth: "170px",
    minWidth: "110px",
    padding: "6px 8px",
  },
})
