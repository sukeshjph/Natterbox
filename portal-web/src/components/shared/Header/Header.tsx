import React from "react"
import Paper from "@material-ui/core/Paper"
import Tabs from "@material-ui/core/Tabs"
import Tab from "@material-ui/core/Tab"
import AppBar from "@material-ui/core/AppBar"
import InputBase from "@material-ui/core/InputBase"
import Icon from "@mdi/react"
import { mdiMagnify } from "@mdi/js"
import { NavLink } from "react-router-dom"
import { tabStyles, tabsStyles } from "./HeaderMUIStyles"
import { Login } from "../Login/Login"
import styles from "./Header.module.scss"
import { AccountIcon } from "../Images"

type headerProps = {
  location: {
    pathname: string
  }
}

const PagesLinks = {
  Home: "/home",
  Logs: "/logs",
  "Account settings": "/accounts",
  Users: "/users",
  Groups: "/groups",
  Devices: "/devices",
  Numbers: "/numbers",
  Policies: "/policies",
}

export const Header = ({ location }: headerProps) => {
  const tabClasses = tabStyles()
  const tabsClasses = tabsStyles()

  return (
    <AppBar position="static" className={styles.root}>
      <div id="header-main" className={styles.header}>
        <div className={styles.upperHeader}>
          <div className={styles.buttonGreen}>
            <AccountIcon />
            Brightec
          </div>
          <h2>Natterbox</h2>
          <Login />
        </div>
        <div className={styles.lowerHeader}>
          <Tabs
            value={location.pathname}
            classes={{
              root: tabsClasses.root,
            }}>
            {Object.keys(PagesLinks).map(pageKey => (
              <Tab
                label={pageKey}
                component={NavLink}
                to={PagesLinks[pageKey]}
                className={`${styles.tabLink} ${
                  location.pathname === PagesLinks[pageKey]
                    ? `${styles.tabLinkSelected}`
                    : ""
                }`}
                classes={{
                  root: tabClasses.root,
                }}
              />
            ))}
          </Tabs>
          <Paper className={styles.headerSearch}>
            <InputBase
              className={styles.inputTextWithIconLarge}
              placeholder="Search Orgâ€¦"
              inputProps={{ "aria-label": "search" }}
            />
            <div>
              <Icon path={mdiMagnify} size={1} />
            </div>
          </Paper>
        </div>
      </div>
    </AppBar>
  )
}
