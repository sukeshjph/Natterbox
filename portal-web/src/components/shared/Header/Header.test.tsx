import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import { Header } from "./Header"

describe("Header", () => {
  it("Renders Header", () => {
    const headerProps = {
      location: {
        pathname: "/logs",
      },
    }
    const wrapper = shallow(<Header {...headerProps} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
