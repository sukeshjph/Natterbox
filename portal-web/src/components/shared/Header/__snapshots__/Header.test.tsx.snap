// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders Header 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="root"
  position="static"
>
  <div
    id="header-main"
  >
    <div
      className="upperHeader"
    >
      <div
        className="rectangle"
      >
        <AccountIcon />
        Brightec
      </div>
      <h2>
        Natterbox
      </h2>
      <Login />
    </div>
    <div
      className="lowerHeader"
    >
      <WithStyles(ForwardRef(Tabs))
        classes={
          Object {
            "root": "makeStyles-root-2",
          }
        }
        value="/logs"
      >
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Home"
          to="/home"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink tabLinkSelected"
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Logs"
          to="/logs"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Account settings"
          to="/accounts"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Users"
          to="/users"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Groups"
          to="/groups"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Devices"
          to="/devices"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Numbers"
          to="/numbers"
        />
        <WithStyles(ForwardRef(Tab))
          className="tabLink "
          classes={
            Object {
              "root": "makeStyles-root-1",
            }
          }
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          label="Policies"
          to="/policies"
        />
      </WithStyles(ForwardRef(Tabs))>
      <WithStyles(ForwardRef(Paper))
        className="search"
      >
        <WithStyles(ForwardRef(InputBase))
          inputProps={
            Object {
              "aria-label": "search",
            }
          }
          placeholder="Searchâ€¦"
        />
        <div>
          <Icon
            color={null}
            horizontal={false}
            path="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
            rotate={0}
            size={1}
            spin={false}
            vertical={false}
          />
        </div>
      </WithStyles(ForwardRef(Paper))>
    </div>
  </div>
</WithStyles(ForwardRef(AppBar))>
`;
